org-ql-view.el

```diff
@@ -1032,54 +1032,91 @@ current buffer.  Otherwise BUFFERS-FILES is returned unchanged."
                                (string (expand-file-name it))
                                (otherwise it))
                              list)))
-    ;; TODO: Test this more exhaustively.
-    (pcase buffers-files
-      ((pred listp)
-       (pcase (expand-files buffers-files)
-         ((pred (seq-set-equal-p (mapcar #'expand-file-name (org-agenda-files))))
-          "org-agenda-files")
-         ((and (guard (file-exists-p org-directory))
-               (pred (seq-set-equal-p (org-ql-search-directories-files
-                                       :directories (list org-directory)))))
-          "org-directory")
-         (_ buffers-files)))
-      ((pred (equal (current-buffer)))
-       "buffer")
-      ((or 'org-agenda-files '(function org-agenda-files))
-       "org-agenda-files")
-      ((and (pred bufferp) (guard (buffer-file-name buffers-files)))
-       (buffer-file-name buffers-files))
-      (_ buffers-files))))
+    (-->
+        ;; TODO: Test this more exhaustively.
+        (pcase buffers-files
+          ((pred listp)
+           (pcase (expand-files buffers-files)
+             ((pred (seq-set-equal-p (mapcar #'expand-file-name (org-agenda-files))))
+              "org-agenda-files")
+             ((and (guard (file-exists-p org-directory))
+                   (pred (seq-set-equal-p (org-ql-search-directories-files
+                                           :directories (list org-directory)))))
+              "org-directory")
+             (_ buffers-files)))
+          ((pred (equal (current-buffer)))
+           "buffer")
+          ((or 'org-agenda-files '(function org-agenda-files))
+           "org-agenda-files")
+          ((and (pred bufferp) (guard (buffer-file-name buffers-files)))
+           (buffer-file-name buffers-files))
+          ((pred bufferp)
+           (buffer-name buffers-files))
+          (_ buffers-files))
+      ;; All items needs to be strings to pick duplicates when used with the extend conterpart.
+      ;; So making sure the buffers are convered to file names
+      (if (stringp it)
+          it
+        (-map
+         (lambda (buffer-file)
+           (if (bufferp buffer-file)
+               (--if-let (buffer-file-name buffer-file)
+                   it
+                 (buffer-name buffer-file))
+             buffer-file))
+         it)))))
```

Comment:
This form is very hard to follow with its deep nesting and multiple levels of `if`.  Please rewrite it to be flatter, e.g. using `cond`, `cl-typecase`, etc.